iroha_p2p/src/network.rs

```diff
@@ -31,18 +35,27 @@ where
     E: Encryptor + Send + 'static,
 {
     /// Creates a network structure, that will hold connections to other nodes.
-    pub fn new(broker: Broker /*handler: AlwaysAddr<A>*/) -> Self {
-        Self {
+    pub fn new(
+        broker: Broker,
+        listen_addr: String, /*handler: AlwaysAddr<A>*/
+    ) -> Result<Self, Error> {
+        let listen_addr: SocketAddr = listen_addr.parse()?;
+        Ok(Self {
             peers: HashMap::new(),
+            listen_addr,
```

Comment:
You definitely can open listener here. Address is not state of network

Reply:
Moved TcpListener start to constructor.